doc/contribute/devtheory.rst

```diff
@@ -202,3 +202,112 @@ the known current volume, the next volume is determined by looking at the other
 volume tagged on that surface (as described in the Sense Tags section above).
 This assumes that a valid surface and volume are provided. If no next volume
 exists, then the call will return 0 for the next volume.
+
+DAGMC Metadata
+~~~~~~~~~~~~~~
+
+DAGMC's Metadata Interface
+--------------------------
+
+DAGMC's metadata interface allows one to navigate metadata in more
+straightforward ways, allowing one to retrieve entities with specific
+attributes quickly. The metadata class is constructed using an existing DAGMC
+class. Upon calling ``load_property_data``, the interface will parse all
+metadata existing in the MOAB file with keywords that have been provided to the
+interface (a list of the default keywords is provided below). Additional
+keywords can be provided to the interface during its construction to support
+implementation-specific conventions.
+
+Default metadata keyword list:
+  - mat: material composition assignment
+  - rho: material density assignment (units: g/cc)
+  - boundary: boundary condition
+  - tally: indicates a geometric tally on a surface or volume
+- importance: used to set variance reduction properties [1]_
+
+The ``mat`` and ``rho`` keywords are required for all DAGMC simulations for full
+material definitions in each volume of the model.
+
+Once the metadata in the MOAB file has been parsed, the interface contains
+functions like ``get_surface_property`` and ``get_volume_property`` which allows
+one to retrieve metadata values for a keyword using the surface/volume index or
+MOAB entity handle.
+
+Metadata Structure in DAGMC Files
+---------------------------------
+
+This section describes the structure used to represent DAGMC metadata in
+MOAB. This is useful knowledge when constructing DAGMC geometries manually or
+when adding support for their generation from a new source.
+
+DAGMC metadata used during simulation (material assignments, boundary
+conditions, volume tallies, etc.) is stored on a MOAB entity set tagged with the
+metadata information using a `NAME` tag. Please see the `UWUW Section
+<../usersguide/uw2.html>`_ of the user's guide for information about the syntax
+of this tagged information. Geometric entities associated with this metadata
+entry are contained by the metadata entity set.
+
+Metadata entity sets are also tagged with a `CATEGORY` tag (similar to geometric
+sets) with the value "Group" to indicate that the entity set's purpose is to
+group geometric entities together. These "Group" sets can be gathered using this
+tag and value to identify "Group" entity sets in the MOAB instance.
+
+.. [1] only used in DAG-MCNP simulations.
+
+The University of Wisconsin Unified Workflow (UWUW) Interface
```

Comment:
How does the metadata API interact with the UWUW API?  (If they do...) That's not clear to me.

Reply:
I'll have to look into that. I think they do. I know the data model in MOAB quite well, but I think @makeclean knows most about these classes and how they connect. I'll take a look at the code and double check my understanding with him then submit some updates.

Reply:
Not looking for too much - I don’t want to hold it up too much